sklearn/metrics/cluster/_supervised.py

```diff
@@ -335,17 +335,45 @@ def adjusted_rand_score(labels_true, labels_pred):
     Parameters
     ----------
     labels_true : int array, shape = [n_samples]
-        Ground truth class labels to be used as a reference
+        Ground truth class labels to be used as a reference.
 
     labels_pred : array-like of shape (n_samples,)
-        Cluster labels to evaluate
+        Cluster labels to evaluate.
 
     Returns
     -------
     ARI : float
        Similarity score between -0.5 and 1.0. Random labelings have an ARI
        close to 0.0. 1.0 stands for perfect match.
+    
+    See Also
+    --------
+    adjusted_mutual_info_score : Adjusted Mutual Information.
+    
+    (tn, fp), (fn, tp) = pair_confusion_matrix(labels_true, labels_pred)
+    # convert to Python integer types, to avoid overflow or underflow
+    tn, fp, fn, tp = int(tn), int(fp), int(fn), int(tp)
+
+    # Special cases: empty data or full agreement
+    if fn == 0 and fp == 0:
+        return 1.0
+
+    return 2.0 * (tp * tn - fn * fp) / ((tp + fn) * (fn + tn) + (tp + fp) * (fp + tn))
```

Comment:
This is not part of the docstring, this is actually the main body of the function itself and hence needs to be below and outside the docstring. If you notice, as you have it right now the function has no code, just the docstring itself.